File: app/src/main/java/com/buzzybrains/mvvmarchitecture/workmanager/SyncWorkManager.java
package com.buzzybrains.mvvmarchitecture.workmanager;

import android.app.Application;
import android.content.Context;

import com.buzzybrains.mvvmarchitecture.data.NoteRepository;
import com.buzzybrains.mvvmarchitecture.model.Note;

import java.util.List;

import androidx.annotation.NonNull;
import androidx.work.Worker;
import androidx.work.WorkerParameters;

public class SyncWorkManager extends Worker {

    NoteRepository noteRepository;

    public SyncWorkManager(@NonNull Context context, @NonNull WorkerParameters workerParams) {
        super(context, workerParams);

        noteRepository = new NoteRepository((Application) getApplicationContext());
    }

    @NonNull
    @Override
    public Result doWork() {

        List<Note> asyncNotes = noteRepository.getAsyncNotes();

        for (Note note : asyncNotes) {
            noteRepository.syncNoteData(note);
        }

        return Result.success();
    }
}

File: app/src/main/java/com/buzzybrains/mvvmarchitecture/workmanager/SyncWorkManager.kt
package com.buzzybrains.mvvmarchitecture.workmanager

import android.app.Application
import android.content.Context
import androidx.work.ListenableWorker

import com.buzzybrains.mvvmarchitecture.data.NoteRepository
import androidx.work.Worker
import androidx.work.WorkerParameters

class SyncWorkManager(context: Context, workerParams: WorkerParameters) : Worker(context, workerParams) {

    internal var noteRepository: NoteRepository

    init {

        noteRepository = NoteRepository(applicationContext as Application)
    }

    override fun doWork(): ListenableWorker.Result {

        val asyncNotes = noteRepository.asyncNotes

        for (note in asyncNotes) {
            noteRepository.syncNoteData(note)
        }

        return ListenableWorker.Result.success()
    }
}
